aten/src/ATen/cpu/vec/vec256/vec256_qint.h

```diff
@@ -1338,4 +1338,32 @@ Vectorized<c10::quint8> inline maximum(const Vectorized<c10::quint8>& a, const V
 }
 
 #endif // if defined(CPU_CAPABILITY_AVX2)
+
+#if defined(CPU_CAPABILITY_NEON)
+template <typename T>
+typename std::enable_if_t<std::is_same_v<T, int8_t>, at::vec::Vectorized<float>>
+inline convert_int8_to_float(at::vec::Vectorized<T> src) {
+  // Note: this function only convert inputs number of elements equal to at::vec::Vectorized<float>.size()
+    auto s8x8 = vld1_s8(src.operator const int8_t*());
+    auto s16x8 = vmovl_s8(s8x8);
+
+    auto s32x4_hi = vmovl_s16(vget_high_s16(s16x8));
+    auto s32x4_lo = vmovl_s16(vget_low_s16(s16x8));
+
+    return Vectorized<float>(vcvtq_f32_s32(s32x4_lo), vcvtq_f32_s32(s32x4_hi));
+}
+
+template <typename T>
+typename std::enable_if_t<std::is_same_v<T, uint8_t>, at::vec::Vectorized<float>>
+inline convert_int8_to_float(at::vec::Vectorized<T> src) {
+  // Note: this function only convert inputs number of elements equal to at::vec::Vectorized<float>.size()
+    auto u8x8 = vld1_u8(src.operator const uint8_t*());
+    auto u16x8 = vmovl_u8(u8x8);
+    auto u32x4_hi = vmovl_u16(vget_high_u16(u16x8));
+    auto u32x4_lo = vmovl_u16(vget_low_u16(u16x8));
```

Comment:
Not sure how much perf it will improve but you can do u8x16 load instead of u8x8 and then extract low and high halves to 16 bits. Then use `vcvtaq_s16_f16` to convert to fp16 and convert that to fp32. TBH though, not sure how much perf gain you will have. Main reason to try this out is 1) fewer loads and 2) use int16 to fp16 conversion and then upconvert.